### DISCUSSION TITLE
bit perfect data streams test (if you care about fidelity)

### DISCUSSION THREAD

**Discussion by fcriver** [mikebrady/shairport-sync#1482] [General]
Hi,

With my external DAC (Naim DAC V1) I can check if a music player streams the data bit perfect.
For that, I have to play a specific WAV file (one for sample rate), and the DAC checks wether the data is bit perfect or not.

In a DELL Latitude E5250 running a Lubuntu, shairport-sync data is bit perfect in these conditions:

ALSA
output_device = "hw:1";             // 1 in my setup                        
ignore_volume_control="yes"
disable_synchronization = "yes"

I listen to only one AirPlay client (running on the Dell Laptop connect to the Naim DAC V1, Classé Amp and ProAc Reference Loudspeakers + Sennheiser HD600 ). So, I don't need synchronization.
No synchronization, no inserting or dropping frames.

Because I have my dose of OCD, I run a strip down Lubuntu without PuseAudio (just ALSA).
I'm not shure if I have to disable synchronization.

Anyway: bit perfect at 16 bit 44.1 KHz from Windows iTunes, Mac Apple Music and JRiver, and third party apps for streaming to apple Air Play devices.

F River

**Comment by mikebrady** [comment]
Thanks, yeah, it should be bit perfect under those conditions. However, with synchronisation off, the buffers will eventually either overflow or empty.

**Reply by fcriver** [reply]
The Naim DAC V1 (if enable) shows in real time, errors, buffer level, and other info related to USB input. 
No errors of any kind, playing long sessions.

**Reply by mikebrady** [reply]
Great. If the source and the sink clocks are really very close, then it is possible. I'm glad it's working so well for you.

**Reply by mikebrady** [reply]
Just looking at the Naim V1 spec, it might just be adapting to the actual rate of data coming from Shairport Sync, which would be fantastic.
